src/Shell/Task/CommandTask.php

```diff
@@ -109,15 +110,22 @@ protected function _scanDir($dir)
     public function commands()
     {
         $shellList = $this->getShellList();
+        $flatten = Hash::flatten($shellList);
+        $duplicates = array_intersect($flatten, array_unique(array_diff_key($flatten, array_unique($flatten))));
```

Comment:
It does not really have any influence actually. I just put it because the ``array_diff_key`` might return the same value multiple times (if there are three shells having the same name for instance, one in the app, one in a plugin and the last one in another plugin).
No need to have the same value multiple times to calculate the intersection.

But I can remove it.